core/opaevaluator.go

```diff
@@ -444,3 +448,32 @@ func LoadRegoModule(rootDirectory string) (*OPAModuleConfig, error) {
 		Content: string(fileContent),
 	}, nil
 }
+
+// verifyAllowed replicates the ResultSet.Allowed function with some slight differences.
+// Since we allow for non boolean return values we use the type assertion to understand
+// whether the returned value is an actual boolean and use it, otherwise we verify the
+// returned value is at least a set containing something, if that's the case we assume
+// this is a custom payload for a response policy and return true regardless.
+//
+// NOTE: do not rely on this function for decision-making conditions; use it only for
+// debugging and informative logging!
+//
+// cfr: https://pkg.go.dev/github.com/open-policy-agent/opa/rego#ResultSet.Allowed
+func verifyAllowed(rs rego.ResultSet) bool {
```

Comment:
Why do not use the double check in the evaluate function?
So, it is true if results.Allowed is true or if there is a response policy (that is if passes the actual check in evaluation response). 
If not allowed, it is possible to set a custom error message anyhow? Maybe it should be the test case, instead of an empty expected body

Reply:
I extracted a func to process the results and prevented doing the results checks twice